### DISCUSSION TITLE
Exporting the whole iMessage database?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [ReagentX/imessage-exporter#317] [Q&A] [answered]
I don't want to export to text or HTML, but make backups of the raw iMessage database (and attachments) stored on my Mac. And would want to export them to another format like text or HTML later.

**Comment by ReagentX** [comment, marked as answer]
Just copy the database file and attachments folder from `~/Library/Messages`, or store a copy of your iOS backup somewhere.

**Reply by YousufSSyed** [reply]
If I made a backup of `~/Library/Messages`, then _much later_ moved the current version of the folder to a temporary directory, and put the backup in `~/Library`, would iMessage work fine?

I'm wondering about the integrity of the messages folder: the ability to see all of that data later and for it to not be corrupt if I backed it up.

**Reply by ReagentX** [reply]
I don't know how Messages.app would handle that.

**Reply by YousufSSyed** [reply]
Is there quickly view messages in all conversations like in iMessage without the app? I tried opening `chat.db` (24MB) with Table Plus but it didn't show anything whatsoever inside.

**Reply by ReagentX** [reply]
No, that's why I build the exporter.

**Reply by YousufSSyed** [reply]
It seems like its fine to backup the folder so I'll marked this as answered.
